### PR TITLE
feat: support MongoDB as a destination

### PR DESCRIPTION
**Pull request by lucasmarshall**
## Test Plan

Tested locally against MongoDB Atlas

### CODE REVIEW

File: openapi/v2/mgmt/components/schemas/create_update_destination.yaml [hunk 1]
```diff
@@ -1,45 +1,65 @@
-allOf:
+oneOf:
```
Comment (albertyfwu) [inline openapi/v2/mgmt/components/schemas/create_update_destination.yaml:1]: why is this necessary? is there a bug with the generated code?

File: openapi/v2/mgmt/components/schemas/objects/destination.yaml [hunk 2]
```diff
@@ -1,5 +1,6 @@
-allOf:
+oneOf:
```
Comment (albertyfwu) [inline openapi/v2/mgmt/components/schemas/objects/destination.yaml:1]: same comment here

File: openapi/v2/mgmt/components/schemas/objects/mongodb_config.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+type: object
+properties:
+  host:
+    type: string
+    example: mongodb.company.com
+  database:
```
Comment (albertyfwu) [inline openapi/v2/mgmt/components/schemas/objects/mongodb_config.yaml:6]: port is missing here

File: packages/core/destination_writers/mongodb.ts [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+
+            callback(null, mappedRecord);
+          } catch (e: any) {
+            return callback(e);
+          }
+        },
+      }),
+      stream
+    );
+    childLogger.info({ collectionName }, 'Importing common object objects into collection [COMPLETED]');
```
Comment (albertyfwu) [inline packages/core/destination_writers/mongodb.ts:150]: not just scoped to this PR, but i noticed this `common object objects` in a bunch of places. could we replace them all with `common object records`?

File: openapi/v2/mgmt/components/schemas/create_update_destination.yaml (hunk 1 above)
Comment (lucasmarshall) [reply openapi/v2/mgmt/components/schemas/create_update_destination.yaml:1]: Docs don't generate correctly unless we do it this way. This is the issue that was called out last week.

File: openapi/v2/mgmt/components/schemas/objects/destination.yaml (hunk 2 above)
Comment (lucasmarshall) [reply openapi/v2/mgmt/components/schemas/objects/destination.yaml:1]: Docs don't generate correctly unless we do it this way. This is the issue that was called out last week.

File: openapi/v2/mgmt/components/schemas/objects/mongodb_config.yaml (hunk 3 above)
Comment (lucasmarshall) [reply openapi/v2/mgmt/components/schemas/objects/mongodb_config.yaml:6]: Port is not supported by MongoDB Atlas, and that's all we are supporting for now.

File: docs/docs/destinations/mongodb.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+| ------------------------------------ | --------- |
+| Syncs: Full refresh                  | Yes       |
+| Syncs: Incremental refresh           | Yes       |
+| Syncs: Supports incremental deletes  | Yes       |
+| Cache invalidation on write          | Yes       |
+| Common objects (unified by Supaglue) | Yes       |
+| Standard objects (per provider)      | Yes       |
+| Custom objects (per provider)        | Yes       |
+
+## Setup
```
Comment (tomkit) [inline docs/docs/destinations/mongodb.md:23]: Can you go through the steps for users to provide a separate account for Supaglue's purposes

File: openapi/v2/mgmt/components/schemas/objects/mongodb_config.yaml (hunk 3 above)
Comment (albertyfwu) [reply openapi/v2/mgmt/components/schemas/objects/mongodb_config.yaml:6]: what i mean is that it's listed under `required` but isn't present here

File: packages/core/destination_writers/mongodb.ts (hunk 4 above)
Comment (lucasmarshall) [reply packages/core/destination_writers/mongodb.ts]: Done

File: docs/docs/destinations/mongodb.md (hunk 5 above)
Comment (lucasmarshall) [reply docs/docs/destinations/mongodb.md:23]: Done.

File: openapi/v2/mgmt/components/schemas/objects/mongodb_config.yaml (hunk 3 above)
Comment (lucasmarshall) [reply openapi/v2/mgmt/components/schemas/objects/mongodb_config.yaml:6]: Good catch. It didn't seem to cause an issue with docs or API validation, but I will remove it.

**Review by albertyfwu — APPROVED** [review]
